packages/edit-site/src/store/selectors.js

```diff
@@ -170,3 +170,58 @@ export function isNavigationOpened( state ) {
 export function isInserterOpened( state ) {
 	return state.blockInserterPanel;
 }
+
+/**
+ * Returns the default template types.
+ *
+ * @param {Object} state Global application state.
+ *
+ * @return {Object} The template types.
+ */
+export function getDefaultTemplateTypes( state ) {
+	return state.settings?.defaultTemplateTypes;
+}
+
+/**
+ * Returns a default template type searched by slug.
+ *
+ * @param {Object} state Global application state.
+ * @param {string} slug The template type slug.
+ *
+ * @return {Object} The template type.
+ */
+export function getDefaultTemplateType( state, slug ) {
+	// Slugs are strings, but numeric template types are encoded as numbers in JS.
+	const parsedSlug = ! isNaN( Number( slug ) ) ? Number( slug ) : slug;
```

Comment:
This is annoying, but I don't see a cleaner way around it.

When the default template types list in PHP is encoded to JSON to be fed to Redux, numeric slugs (such as `404`) are converted to numbers.

But slugs in WP are strings, and are returned as such by the API.
Then, `find` strictly compare the values (e.g. `404 === "404"`) and fails.

I could have checked just for `404`, but I think we might have other templates with numeric slugs.
For example, a plugin could create a custom template with slug `12345`, and then provide details for it by filtering the default template types list.

Reply:
> This is annoying,

🤣  - `! isNaN( Number( slug ) ) ?` - a bit annoying, yes!  But I can't think of a better way to go about it.

> I could have checked just for `404`

As you pointed out, `isNaN( Number( slug ) )` will handle more potential cases in the future and seems like the right way to go.

Reply:
I'm not sure I'm following this discussion.

> When the default template types list in PHP is encoded to JSON to be fed to Redux, numeric slugs (such as 404) are converted to numbers.

Is this not a code smell? Can you point to where this is happening?

Reply:
@mcsf The short answer is that PHP casts numeric keys to `int`, which eventually propagated all the way to JS, messing up the comparison between the slug returned by the API (string) and the one contained in the default template types list (int).

Regardless, I gave the code another look, and realized this could have been solved in PHP by simply casting the keys to string when converting the list to JS usage*.
See d19efb6

---

*: additional context about the difference between the PHP and JS lists.

PHP uses an associative array with slugs as keys, while JS an array of objects with slugs as properties.
The reason for this difference is that JS would reorder numeric object properties to the top, practically always showing the `404` template first.

On the other hand, associative arrays in PHP are massively easier to handle (e.g. finding an element by one of its property is trivial in JS, but annoying in PHP).
Since this array is filterable, I'd rather have it easier to handle than more consistent between PHP and JS (which imho is not a big deal anyway).

Reply:
Thanks for circling back to this, it looks much better.